File: Aegis-Nexus/backend/models/event.py
from dataclasses import dataclass, field
from datetime import datetime, timezone
from typing import Literal

EventType = Literal[
    "SERVICE_STARTED",
    "THRESHOLD_CROSSED",
    "RECOVERY",
    "SERVICE_UNAVAILABLE",
    "TELEMETRY",
]


def utc_now() -> datetime:
    return datetime.now(timezone.utc)


@dataclass(slots=True)
class TimelineEvent:
    id: str
    service_id: str
    service_name: str
    type: EventType
    severity: str
    message: str
    timestamp: datetime = field(default_factory=utc_now)

File: Aegis-Nexus/frontend/types/event.ts
export type TimelineEventType =
  | "SERVICE_STARTED"
  | "THRESHOLD_CROSSED"
  | "RECOVERY"
  | "SERVICE_UNAVAILABLE"
  | "TELEMETRY";

export interface TimelineEvent {
  id: string;
  service_id: string;
  service_name: string;
  type: TimelineEventType;
  severity: "info" | "success" | "warning" | "critical";
  message: string;
  timestamp: string;
}
